modular_zubbers/code/modules/antagonists/bloodsucker/powers/celerity.dm

```diff
@@ -0,0 +1,123 @@
+#define CELERITY_FX_LEVEL 3
+#define CELERITY_DODGE_LEVEL 6
+#define CELERITY_BLUR "celerity_blur"
+
+/datum/action/cooldown/bloodsucker/celerity
+	name = "Celerity"
+	desc = "Grant yourself supernatural speed and reflexes."
+	button_icon_state = "power_speed"
+	power_flags = BP_CONTINUOUS_EFFECT
+	check_flags = AB_CHECK_CONSCIOUS
+	purchase_flags = BLOODSUCKER_CAN_BUY|GHOUL_CAN_BUY
+	bloodcost = 10
+	constant_bloodcost = 1
+	cooldown_time = 20 SECONDS
+
+/datum/action/cooldown/bloodsucker/celerity/get_power_explanation_extended()
+	. = list()
+	. += "Utilize your blood to increase your speed and reflexes."
+	. += "This will drain blood at a constant rate of one unit per second while active."
+	. += "You'll be able to move faster and attack faster, scaling with level."
+	. += "At level [CELERITY_FX_LEVEL], you'll move fast enough that you can no longer conceal your Celerity."
+	. += "At level [CELERITY_DODGE_LEVEL], you'll be able to dodge projectiles for the first few seconds of your Celerity."
+
+/datum/action/cooldown/bloodsucker/celerity/ActivatePower(atom/target)
+	var/mob/living/user = owner
+	var/datum/movespeed_modifier/celerity_mod = new()
+	celerity_mod.multiplicative_slowdown = -0.05*level_current //lightpink extract at level 10, just without the pacifism
+	owner.add_movespeed_modifier(celerity_mod, update = TRUE)
```

Comment:
cache the modifier like fortitude does to prevent issues arising

Reply:
might not be neccesary given we deactivate on levelup, but It still feels worth it, up to you.